services/user/Fractals/service/src/tables.rs

```diff
@@ -0,0 +1,573 @@
+
+            Self {
+                account,
+                created_at: now,
+                mission,
+                name,
+                reward_wait_period: 86400 * 7 * 1,
+            }
+        }
+
+        pub fn add(account: AccountNumber, name: String, mission: String) {
+            Self::new(account, name, mission).save();
+        }
+
+        pub fn get(fractal: AccountNumber) -> Option<Self> {
+            let table = FractalTable::new();
+            table.get_index_pk().get(&(fractal))
+        }
+
+        pub fn get_assert(fractal: AccountNumber) -> Self {
+            check_some(Self::get(fractal), "fractal does not exist")
+        }
+
+        pub fn save(&self) {
+            let table = FractalTable::new();
+            table.put(&self).expect("failed to save");
+        }
+
+        pub fn members(&self) -> Vec<Member> {
+            let table = MemberTable::new();
+            table
+                .get_index_pk()
+                .range(
+                    (self.account, AccountNumber::new(0))
+                        ..=(self.account, AccountNumber::new(u64::MAX)),
+                )
+                .collect()
+        }
+    }
+
+    #[derive(PartialEq)]
+    pub enum MemberStatus {
+        Visa = 1,
+        Citizen = 2,
+        Exiled = 3,
+    }
+
+    pub type StatusU32 = u32;
+
+    impl From<StatusU32> for MemberStatus {
+        fn from(status: StatusU32) -> Self {
+            match status {
+                1 => MemberStatus::Visa,
+                2 => MemberStatus::Citizen,
+                3 => MemberStatus::Exiled,
+                _ => abort_message("invalid member status"),
+            }
+        }
+    }
+
+    #[table(name = "MemberTable", index = 1)]
+    #[derive(Default, Fracpack, ToSchema, SimpleObject, Serialize, Deserialize, Debug)]
+    pub struct Member {
+        pub fractal: AccountNumber,
+        pub account: AccountNumber,
+        pub created_at: psibase::TimePointSec,
+        pub member_status: StatusU32,
+
+        pub reward_balance: u64,
+        pub reward_start_time: psibase::TimePointSec,
+        pub reward_wait: u32,
+    }
+
+    impl Member {
+        #[primary_key]
+        fn pk(&self) -> (AccountNumber, AccountNumber) {
+            (self.fractal, self.account)
+        }
+
+        #[secondary_key(1)]
+        fn by_member(&self) -> (AccountNumber, AccountNumber) {
+            (self.account, self.fractal)
+        }
+
+        fn new(fractal: AccountNumber, account: AccountNumber, status: MemberStatus) -> Self {
+            let now = TransactSvc::call().currentBlock().time.seconds();
+            Self {
+                account,
+                fractal,
+                created_at: now,
+                member_status: status as StatusU32,
+                reward_balance: 0,
+                reward_start_time: now,
+                reward_wait: 0,
+            }
+        }
+
+        fn deposit(&mut self, amount: u64) {
```

Comment:
`deposit`, `withdraw`, and `withdrawable_amount` are unused

Reply:
Consider removing until a later date and delivering the reward system all at once